File: designPattern/Entry/Basic/src/Directory.java
import java.util.ArrayList;
import java.util.Iterator;

public class Directory extends Entry{
    private String name;
    private ArrayList directory = new ArrayList();

    public Directory(String name){
        this.name = name;
    }
    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getSize() {
        int size = 0;
        Iterator it = directory.iterator();

        while(it.hasNext()){
            Entry entry = (Entry)it.next();
            size += entry.getSize();
        }
        return size;
    }

    public Entry add(Entry entry){
        directory.add(entry);
        return this;
    }

    @Override
    protected void printList(String prefix) {
        System.out.println(prefix+"/"+this);
        Iterator iterator = directory.iterator();
        while(iterator.hasNext()){
            Entry entry = (Entry)iterator.next();
            entry.printList(prefix+"/"+name);
        }
    }
}

File: designPattern/Composite/basic2/Directory.py
from Entry import Entry


class Directory(Entry):

    def __init__(self, name):
        self.name = name
        self.array = []

    def print_path(self, prefix):
        print(prefix+"/"+super().__repr__())
        for fileDirec in self.array:
            fileDirec.print_path(prefix+"/"+self.name)

    def add(self, entry):
        self.array.append(entry)

    def get_name(self):
        return self.name

    def get_size(self):
        size = 0
        for fileDirec in self.array:
            size += fileDirec.get_size()

        return size
